lib/interface-types/src/interpreter/wasm/structures.rs

```diff
@@ -0,0 +1,157 @@
+use super::values::{InterfaceType, InterfaceValue};
+use std::{cell::Cell, ops::Deref};
+
+pub trait TypedIndex: Copy + Clone {
+    fn new(index: usize) -> Self;
+    fn index(&self) -> usize;
+}
```

Comment:
It would be good to document the reason why we can't use the runtime-core version of this trait!

Reply:
Because `wasmer-interface-types` is designed to be runtime agnostic, so with no connection with Wasmer (except in the name), so that it can be reused by other runtimes (I'm probably dreaming here 😇).